### PR TITLE
c-ares: add version 1.30.0

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **c-ares/1.30.0**

https://github.com/c-ares/c-ares/compare/cares-1_28_1...v1.30.0

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by toge** [comment]
There is a release bug in upstream.
https://github.com/c-ares/c-ares/issues/763

Waiting for fixed release.

File: recipes/c-ares/all/test_package/test_package.c [hunk 1]
```diff
@@ -103,7 +103,6 @@ main(void)
         return 1;
     }
 
-    ares_gethostbyname(channel, "google.com", AF_INET, callback, NULL);
```
Comment (RubenRBS) [inline recipes/c-ares/all/test_package/test_package.c:106]: Yay! Thanks!

**Review by RubenRBS — APPROVED** [review]
<3

**Review by perseoGI — APPROVED** [review]
LGTM

**Review by jcar87 — APPROVED** [review]

**Pull request merged**